include/pm/device.h

```diff
@@ -142,6 +144,9 @@ struct pm_device {
 				DT_NODE_EXISTS(node_id),		\
 				(DT_PROP_OR(node_id, wakeup_source, 0)),\
 				(0)) << PM_DEVICE_FLAG_WS_CAPABLE),	\
+		.domain = COND_CODE_1(DT_NODE_HAS_PROP(node_id, power_domain),	\
+			(DEVICE_GET(Z_DEVICE_DT_DEV_NAME(DT_PROP(node_id, \
```

Comment:
can `DEVICE_DT_GET_OR_NULL(DT_PHANDLE(node_id, power_domain))` can be used here?

Reply:
Yep, this macro was introduce after I've written it. The macro make it simpler indeed.

Reply:
not resolved

Reply:
it was in the commit making it optional.